guide/xml/portfile-subports.xml

```diff
@@ -0,0 +1,49 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE section PUBLIC "-//OASIS//DTD DocBook XML V5.0//EN"
+"http://docbook.org/xml/5.0/dtd/docbook.dtd">
+<section xml:id="reference.subports" xmlns:xlink="http://www.w3.org/1999/xlink">
+  <title>Subports</title>
+
+  <para>The subport declaration causes MacPorts to define an additional port,
+  with the <literal>name</literal> given by the declaration. The keywords
+  for the subport are those in the global section of the Portfile, and those
+  in the brace-enclosed <literal>body</literal>.</para>
+
+  <variablelist>
+    <varlistentry>
+      <term>subport <replaceable>name</replaceable> <replaceable>body</replaceable></term>
+
+      <listitem>
+        <para>Example:</para>
+
+        <programlisting>
+Portfile                   1.0
+
+name                       example
+
+depends_lib                aaa
+configure.args             --bbb
+
+subport example-sub1 {
+    depends_lib-append     ccc
+    configure.args         --ddd
+}
+
+subport example-sub2 {
+    depends_lib-append     eee
+    configure.args-append  --fff
+}</programlisting>
+      </listitem>
+
+      <listitem>
+        <para>Because MacPorts treats each subport as a separate port declaration,
+        each subport will have its own, independent phases: fetch, configure,
+        build, destroot, install, activate, etc. However, because the subports
+        share the global declaration part, all the subports will by default
+        share the same dist_subdir. This means that MacPorts only needs to fetch
+        the distfiles once, and the remaining subports can reuse the
+        distfiles.</para>
```

Comment:
Similarly, `distname` doesn't automatically change in a subport because its default value is `${name}-${version}`. And in fact the same is true for any variable that isn't derived from `$subport`, which is most of them.

But perhaps it should be made clear that a subport *can* explicitly override anything apart from `name` and `subport`, it just doesn't happen implicitly.